metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -76,20 +121,104 @@ def __init__(self, config: KafkaSourceConfig, ctx: PipelineContext):
         )
         self.report = KafkaSourceReport()
 
+    def is_checkpointing_enabled(self, job_id: JobId) -> bool:
+        if (
+            job_id == self.get_default_ingestion_job_id()
+            and self.is_stateful_ingestion_configured()
+            and self.source_config.stateful_ingestion
+            and self.source_config.stateful_ingestion.remove_stale_metadata
+        ):
+            return True
+
+        return False
+
+    def get_default_ingestion_job_id(self) -> JobId:
+        """
+        Default ingestion job name that kafka provides.
+        """
+        return JobId("ingest_from_kafka_source")
+
+    def create_checkpoint(self, job_id: JobId) -> Optional[Checkpoint]:
+        """
+        Create a custom checkpoint with empty state for the job.
+        """
+        assert self.ctx.pipeline_name is not None
+        if job_id == self.get_default_ingestion_job_id():
+            return Checkpoint(
+                job_name=job_id,
+                pipeline_name=self.ctx.pipeline_name,
+                platform_instance_id=self.get_platform_instance_id(),
+                run_id=self.ctx.run_id,
+                config=self.source_config,
+                state=KafkaCheckpointState(),
+            )
+        return None
+
+    def get_platform_instance_id(self) -> str:
+        return (
+            f"{self.source_config.platform_instance}"
+            if self.source_config.platform_instance
```

Comment:
I don't think lines 160 & 161 are not necessary because of line 103. Just `assert self.source_config.platform_instance is not None`.